src/ids-rating/ids-rating.js

```diff
@@ -0,0 +1,209 @@
+import {
+  IdsElement,
+  customElement,
+  scss,
+  mix
+} from '../ids-base/ids-element';
+
+import { IdsEventsMixin } from '../ids-base/ids-events-mixin';
+import { IdsKeyboardMixin } from '../ids-base/ids-keyboard-mixin';
+
+// @ts-ignore
+import styles from './ids-rating.scss';
+
+/**
+ * IDS Rating Component
+ * @type {IdsRating}
+ * @inherits IdsElement
+ * @mixes IdsEventsMixin
+ * @mixes IdsKeyboardMixin
+ */
+
+ @customElement('ids-rating')
+ @scss(styles)
+
+class IdsRating extends mix(IdsElement).with(IdsEventsMixin, IdsKeyboardMixin) {
+   constructor() {
+     super();
+   }
+
+    ratingsConfig = {
+      ratingsAttr: {
+        star: this.getAttribute('stars'),
+        active: this.getAttribute('active'),
+        color: this.getAttribute('color'),
+        size: this.getAttribute('size'),
+        value: this.getAttribute('value'),
+        readonly: this.getAttribute('readonly'),
+        clickable: this.getAttribute('clickable'),
+        compact: this.getAttribute('compact')
+      }
+    }
+
+    connectedCallback() {
+      this.buildDOM();
+      console.log(this.ratingsConfig)
+    }
+
+    /**
+     * Create the Template for the contents
+     * @returns {string} The template
+     */
+    template() {
+      return `<div id="rating"></div>`;
+    }
+
+    /**
+    * @returns {Array<string>} this component's observable properties
+    */
+    static get properties() {
+      return [
+        'value',
+        'stars',
+        'readonly'
+      ];
+    }
+
+    /**
+    * @param {any} v
+    */
+    set value(v) {
+      if(v) {
+        this.setAttribute('value', v.toString())
+      }
+    }
+
+    get value() {
+      return this.getAttribute('value')
+    }
+
+    /**
+    * @param {any} v
+    */
+    set stars(v) {
+      if(v) {
+        this.setAttribute('stars', v.toString());
+      }
+    }
+
+    get stars() {
+      return this.getAttribute('stars')
```

Comment:
```suggestion
      return this.getAttribute('stars') || 5;
```

For the defaults you did above you can just do this....

Reply:
Thanks, I will keep this in mind as I work through it. I was trying to write some defensive logic in case the attribute was not added to the wc. The setter and getter do not fire when the attribute is not present. That was my thinking behind it.